src/components/views/ViewCampaign.jsx

```diff
@@ -110,12 +130,58 @@ const ViewCampaign = ({ match }) => {
     );
   };
 
-  useEffect(() => {
-    const { id, slug } = match.params;
-    const getFunction = slug
-      ? CampaignService.getBySlug.bind(CampaignService, slug)
-      : CampaignService.get.bind(CampaignService, id);
+  const { id, slug } = match.params;
+  const getFunction = slug
+    ? CampaignService.getBySlug.bind(CampaignService, slug)
+    : CampaignService.get.bind(CampaignService, id);
 
+  const updateLeaderboard = () => {
+    loadNewAggregateDonations();
+    getFunction()
+      .then(_campaign => {
+        setCampaign(_campaign);
+      })
+      .catch(() => {
+        setNotFound(true);
+      });
+  };
+
+  const loadDonations = campaignId => {
+    if (campaignId) {
+      CampaignService.getDonations(
+        campaignId,
+        donationsPerBatch,
+        0,
+        (_donations, _donationsTotal) => {
+          const allDonations = [..._donations, ...donations];
+          setDonations(allDonations);
+          const BreakException = {};
+          try {
+            allDonations.forEach(item => {
+              if (item._status === 'Committed') {
+                if (
+                  lastCommittedDonation.current &&
+                  new Date(lastCommittedDonation.current._createdAt) < new Date(item._createdAt)
+                ) {
+                  updateLeaderboard();
```

Comment:
updateLeaderBoard should be called just once new committed/waiting donation is created